packages/core/src/plugins/pluginsRegistry.ts

```diff
@@ -0,0 +1,44 @@
+import type { LoggerSettings } from './loggerSettings';
+import type { MetricsRecorderSettings } from './metricsRecorderSettings';
+
+/**
+ * Internal object to hold all of the plugins.
+ *
+ * @emarks Not exported to encapsulate the registry
+ */
+const pluginsRegistry: {
+  loggerSettings?: LoggerSettings;
+  metricsRecorderSettings?: MetricsRecorderSettings;
+} = {};
+
+/**
+ * Register a Logger for logging AppKit logs.
+ * @param param0 settings for Logger
+ * @alpha
+ */
+export const registerLogger = (settings: LoggerSettings) => {
```

Comment:
Is there some way we can make a generic registerPlugin method? I am thinking we'll end up having a lot of methods in the long run

Reply:
2 ideas, Not sure this will work nicely with typings, 

```
const registerPlugin = (nameSpace: string; plugin: IPlugin) => pluginRegistry[nameSpace] = plugin;
```

```
type Plugin = {
  install: (pluginRegistry: PluginRegistry) => void;
}

const loggerPlugin: Plugin = {
  install: (pluginRegistry) => {
    pluginRegistry.logger = myCustomLogger;
  }
}
```

Reply:
lemme try this 🙏

Reply:
i'm able to get nice typed function for registration; however, I'm not sure how the retrieval would work 🤔

```
// Registration

export type PluginSettings<NameSpace extends string, Plugin> = { namespace: NameSpace; provider: () => Plugin };
export type PluginType<NameSpace extends string, Plugin> = [NameSpace, Plugin];

export type LoggerSettings = PluginSettings<'logger', Logger>;
export type MetricsRecorderSettings = PluginSettings<'metricsRecorder', MetricsRecorder>;

const pluginsRegistry: {
  loggerSettings?: LoggerSettings;
  metricsRecorderSettings?: MetricsRecorderSettings;
} = {};

export const registerPlugin = (settings: LoggerSettings | MetricsRecorderSettings) => {
  if (settings.namespace === 'logger') {
    pluginsRegistry.loggerSettings = settings;
  }
  if (settings.namespace === 'metricsRecorder') {
    pluginsRegistry.metricsRecorderSettings = settings;
  }
};

```

```
// Retrieval

export const getPlugin = <P>(namespace: 'logger' | 'metricsRecorder'): P | undefined => {
  if (namespace === 'logger') {
    return pluginsRegistry.loggerSettings?.provider();
  }
  if (namespace === 'metricsRecorder') {
    return pluginsRegistry.metricsRecorderSettings?.provider();
  }
};

```

How do we provide something to the function and make sure it is type safe? the `getPlugin` above has errors of
```
Type 'Logger | undefined' is not assignable to type 'P | undefined'.
  Type 'Logger' is not assignable to type 'P'.
    'P' could be instantiated with an arbitrary type which could be unrelated to 'Logger'.
```
and 
```
Type 'MetricsRecorder | undefined' is not assignable to type 'P | undefined'.
  Type 'MetricsRecorder' is not assignable to type 'P'.
    'P' could be instantiated with an arbitrary type which could be unrelated to 'MetricsRecorder'.
```